### PR TITLE
NOISSUE - Change file structure and add client and topic management

### PR DESCRIPTION
**Pull request by drasko**
Signed-off-by: Drasko Draskovic <drasko.draskovic@gmail.com>

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+}
+
+// Client stores complete MQTT client data
+type Client struct {
+	info      ClientInfo
+	conn      net.Conn
+	keepalive uint16
+	lwt       *packets.PublishPacket
+	handler   auth.Handler
+	subs      map[string]topic.Subsciption
```
Comment (mteodor) [inline client/client.go]: Subscription

File: client/client.go [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+}
+
+func (c Client) ReadLoop() {
+	keepAlive := time.Second * time.Duration(c.info.keepalive)
+	timeOut := keepAlive + (keepAlive / 2)
+	dpkt := packets.NewControlPacket(packets.Disconnect).(*packets.DisconnectPacket)
+
+	for {
+		select {
+		case <-c.ctx.Done():
```
Comment (mteodor) [inline client/client.go]: `ctx`? I guess you'll add it later

File: client/client.go [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+			return
+		default:
+			if keepAlive > 0 {
+				if err := c.conn.SetReadDeadline(time.Now().Add(timeOut)); err != nil {
+					c.logger.Error("Set read timeout error: ", zap.Error(err), zap.String("client_id", c.info.ID))
```
Comment (manuio) [inline client/client.go]: I would not use `error` in the error message

File: client/client.go [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+
+		// Handle Auth Subscribe
+		if err := c.handler.AuthSubscribe(&c.info); err != nil {
+			s.logger.Error("Auth Subscribe error, ", zap.Error(err), zap.String("client_id", ci.ID))
+			retcodes = append(retcodes, qosFail)
+			continue
+		}
+
+		// Shared subscriptions
+		group := ""
```
Comment (manuio) [inline client/client.go]: `var group string`

File: client/client.go [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+
+// Unsubscribe
+func (c *client) processUnsubscribe(packet *packets.UnsubscribePacket) {
+	if c.status == Disconnected {
+		return
+	}
+	b := c.broker
+	if b == nil {
+		return
+	}
```
Comment (manuio) [inline client/client.go]: Could this be simplified with `if c.broker == nil`?

File: client/client.go (hunk 2 above)
Comment (drasko) [reply client/client.go]: For now this is just a copy of some other code, I need to clean... I posted PR just for early preview.

**Pull request merged**